### PR TITLE
Foreign encode and decode of Fixed

### PR DESCRIPTION
**Pull request by jxv**

### CODE REVIEW

File: src/Data/Fixed.purs [hunk 1]
```diff
@@ -110,6 +111,12 @@ reifyPrecision n f = reifyPrecision (n - 1) (f <<< liftTenTimes) where
 -- | operations.
 newtype Fixed (precision :: Precision) = Fixed BigInt.BigInt
 
+instance decodeFixed :: Decode (Fixed precision) where
+  decode f = map (Fixed <<< BigInt.fromInt) (decode f)
```
Comment (paf31) [inline src/Data/Fixed.purs:115]: This isn't compatible with the `Encode` instance.

Comment (jxv) [reply src/Data/Fixed.purs:115]: Are you referring to Number vs Int?

**Review by paf31 — COMMENTED** [review]
I'm not sure these instances belong in this library, or on `Fixed` at all, since there isn't in my opinion a good canonical choice of encoding. We could encode as the integer representation, or via `Number`. Better to use `newtype`s I think.

File: src/Data/Fixed.purs (hunk 1 above)
Comment (jxv) [reply src/Data/Fixed.purs:115]: On the Haskell side, Aeson transforms `Integer`s to ~`Number`s~ _`Scientific`s_.

**Comment by hdgarrood** [comment]
Both the integer representation and via `Number` are going to be lossy though, right? Have we considered using strings?

**Comment by jxv** [comment]
I have considered `String`s, but it's incompatible. Perhaps, what Phil suggested is the least incorrect -- no instance for Fixed and use `newtype`s.

**Pull request closed without merging**